src/core_plugins/kibana/index.js

```diff
@@ -43,39 +44,40 @@ module.exports = function (kibana) {
           id: 'kibana:discover',
           title: 'Discover',
           order: -1003,
-          url: '/app/kibana#/discover',
+          url: `${kbnBaseUrl}#discover`,
```

Comment:
Nope, that was accidental. I'll switch it back for consistency (seems in most places we link to `#/`).